examples/fun-token/src/lib.rs

```diff
@@ -0,0 +1,502 @@
+            if account.balance < remaining_amount {
+                env::panic(b"Not enough unlocked balance");
+            }
+            account.balance -= remaining_amount;
+
+            // If transferring by escrow, need to check and update allowance.
+            if escrow_account_id != owner_id {
+                let allowance = account.get_allowance(&escrow_account_id);
+                // Checking and updating unlocked balance
+                if allowance < remaining_amount {
+                    env::panic(b"Not enough allowance");
+                }
+                account.set_allowance(&escrow_account_id, allowance - remaining_amount);
+            }
+        }
+
+        self.accounts.insert(&owner_id, &account);
+
+        // Deposit amount to the new owner
+        let mut new_account = self.get_account(&new_owner_id);
+        new_account.balance += amount;
+        self.accounts.insert(&new_owner_id, &new_account);
+    }
+
+    /// Same as `transfer_from` with `owner_id` `predecessor_id`.
+    pub fn transfer(&mut self, new_owner_id: AccountId, amount: Balance) {
+        self.transfer_from(env::predecessor_account_id(), new_owner_id, amount);
+    }
+
+    /// Returns total supply of tokens.
+    pub fn get_total_supply(&self) -> Balance {
+        self.total_supply
+    }
+
+    /// Returns total balance for the `owner_id` account. Including all locked and unlocked tokens.
+    pub fn get_total_balance(&self, owner_id: AccountId) -> Balance {
+        self.get_account(&owner_id).total_balance()
+    }
+
+    /// Returns unlocked token balance for the `owner_id`.
+    pub fn get_unlocked_balance(&self, owner_id: AccountId) -> Balance {
+        self.get_account(&owner_id).balance
+    }
+
+    /// Returns current allowance for the `owner_id` to be able to use by `escrow_account_id`.
+    pub fn get_allowance(&self, owner_id: AccountId, escrow_account_id: AccountId) -> Balance {
+        self.get_account(&owner_id).get_allowance(&escrow_account_id)
+    }
+
+    /// Returns current locked balance for the `owner_id` locked by `escrow_account_id`.
+    pub fn get_locked_balance(&self, owner_id: AccountId, escrow_account_id: AccountId) -> Balance {
+        self.get_account(&owner_id).get_locked_balance(&escrow_account_id)
+    }
+}
+
+impl FunToken {
+    /// Helper method to get the account details for `owner_id`.
+    fn get_account(&self, owner_id: &AccountId) -> Account {
+        self.accounts.get(owner_id).unwrap_or_default()
+    }
+}
+
+#[cfg(feature = "env_test")]
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use near_bindgen::MockedBlockchain;
+    use near_bindgen::{testing_env, Config, VMContext};
+
+    fn alice() -> AccountId {
+        "alice.near".to_string()
+    }
+    fn bob() -> AccountId {
+        "bob.near".to_string()
+    }
+    fn carol() -> AccountId {
+        "carol.near".to_string()
+    }
+
+    fn get_context(predecessor_account_id: AccountId) -> VMContext {
+        VMContext {
```

Comment:
these fixtures seems appear a lot of place, would be helpful to provide as a helper?

Reply:
We can provide some default implementation and just modify details.